From 8b2bfae8719d19d6ecee4c4b29e9e6fb808be79f Mon Sep 17 00:00:00 2001
From: Sven Nierlein <sven@nierlein.de>
Date: Mon, 21 Aug 2017 10:03:07 +0200
Subject: [PATCH] fix non well formed numeric value encountered

php 7 is more picky about numbers. adjust_unit() accepts values like
1024MB and then does arithmetic operations on that which worked before
php 7 but now leads to the error above.

Signed-off-by: Sven Nierlein <sven@nierlein.de>
---
 share/pnp/application/helpers/pnp.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/share/pnp/application/helpers/pnp.php b/share/pnp/application/helpers/pnp.php
index 9c32fce..de6a7f1 100644
--- a/share/pnp/application/helpers/pnp.php
+++ b/share/pnp/application/helpers/pnp.php
@@ -121,6 +121,7 @@ public static function adjust_unit($value,$base=1000,$format='%.3lf'){
         preg_match('/^(-?[0-9\.,]+)\s*(\S?)(\S?)/',$value,$matches);
 
         $mag = 0;
+        $value = $matches[1];
         while ($value >= $base){
             $value /= $base;
             $mag++;
